gadopt/stokes_integrators.py

```diff
@@ -140,14 +149,58 @@ def __init__(self, z, T, approximation, bcs=None, mu=1,
                     self.strong_bcs.append(fd.DirichletBC(self.Z.sub(0).sub(1), value, id))
                 elif bc_type == 'uz':
                     self.strong_bcs.append(fd.DirichletBC(self.Z.sub(0).sub(2), value, id))
+                elif bc_type == 'free_surface':
+                    # Copy free surface information to a new dictionary
+                    free_surface_dict[id] = value
+                    self.free_surface = True
                 else:
                     weak_bc[bc_type] = value
             self.weak_bcs[id] = weak_bc
 
+        if self.free_surface:
+            assert free_surface_dt is not None
```

Comment:
Just a minor one: I think this should probably be a check with a `raise` if `free_surface_dt` isn't provided, to give a more informative exception.

Reply:
Thanks Angus - I've just added a type error in with a short description